pkg/config/configuration.go

```diff
@@ -121,5 +143,69 @@ func LoadKubernetesConfiguration(config, namespace string, operatorClient operat
 	if err != nil {
 		return nil, err
 	}
+
+	err = sortAndValidateSecretsConfiguration(&conf)
+	if err != nil {
+		return nil, err
+	}
+
 	return &conf, nil
 }
+
+// Validate the secrets configuration and sort the allow and deny lists if present.
+func sortAndValidateSecretsConfiguration(conf *Configuration) error {
+	scopes := conf.Spec.Secrets.Scopes
+	set := sets.NewString()
+	for _, scope := range scopes {
+		// validate scope
+		if set.Has(scope.StoreName) {
+			return errors.Errorf("%q storeName is repeated in secrets configuration", scope.StoreName)
+		}
+		if scope.DefaultAccess != "" &&
+			!strings.EqualFold(scope.DefaultAccess, AllowAccess) &&
+			!strings.EqualFold(scope.DefaultAccess, DenyAccess) {
+			return errors.Errorf("defaultAccess %q can be either allow or deny", scope.DefaultAccess)
+		}
+		set.Insert(scope.StoreName)
+
+		// modify scope
+		sort.Strings(scope.AllowedSecrets)
+		sort.Strings(scope.DeniedSecrets)
```

Comment:
```suggestion
		scope.DeniedSecrets = sets.NewString(scope.DeniedSecrets...).List() 
```
to also remove duplicates

Reply:
scope is not a pointer, so cannot change the value directly.